src/main/java/io/github/vcuswimlab/stackintheflow/model/L2H/CustomL2H.java

```diff
@@ -0,0 +1,2062 @@
+        this.V = V;
+
+        this.hyperparams = new ArrayList<Double>();
+        this.hyperparams.add(alpha);
+        this.hyperparams.add(beta);
+        this.hyperparams.add(a0);
+        this.hyperparams.add(b0);
+
+        this.switchPrior = new double[2];
+        this.switchPrior[INSIDE] = a0;
+        this.switchPrior[OUTSIDE] = b0;
+
+        this.sampledParams = new ArrayList<ArrayList<Double>>();
+        this.sampledParams.add(cloneHyperparameters());
+
+        this.treeUpdated = treeUp;
+        this.sampleExact = sampleExact;
+
+        this.BURN_IN = burnin;
+        this.MAX_ITER = maxiter;
+        this.LAG = samplelag;
+        this.REP_INTERVAL = repInterval;
+
+        this.initState = initState;
+        this.pathAssumption = pathAssumption;
+        this.paramOptimized = paramOpt;
+        this.prefix += initState.toString();
+        this.setName();
+
+        if (verbose) {
+            logln("--- folder\t" + folder);
+            logln("--- label vocab:\t" + L);
+            logln("--- word vocab:\t" + V);
+            logln("--- alpha:\t" + MiscUtils.formatDouble(alpha));
+            logln("--- beta:\t" + MiscUtils.formatDouble(beta));
+            logln("--- a0:\t" + MiscUtils.formatDouble(a0));
+            logln("--- b0:\t" + MiscUtils.formatDouble(b0));
+            logln("--- burn-in:\t" + BURN_IN);
+            logln("--- max iter:\t" + MAX_ITER);
+            logln("--- sample lag:\t" + LAG);
+            logln("--- paramopt:\t" + paramOptimized);
+            logln("--- initialize:\t" + initState);
+            logln("--- path assumption:\t" + pathAssumption);
+            logln("--- tree builder:\t" + treeBuilder.getName());
+            logln("--- updating tree?\t" + treeUpdated);
+            logln("--- exact sampling?\t" + sampleExact);
+        }
+    }
+
+    protected void setName() {
+        StringBuilder str = new StringBuilder();
+        str.append(this.prefix)
+                .append("_").append(basename)
+                .append("_K-").append(L)
+                .append("_B-").append(BURN_IN)
+                .append("_M-").append(MAX_ITER)
+                .append("_L-").append(LAG)
+                .append("_opt-").append(this.paramOptimized)
+                .append("_").append(this.pathAssumption);
+        for (double hp : this.hyperparams) {
+            str.append("-").append(MiscUtils.formatDouble(hp));
+        }
+        str.append("-").append(treeBuilder.getName());
+        str.append("-").append(treeUpdated);
+        str.append("-").append(sampleExact);
+        this.name = str.toString();
+    }
+    
+    public Node[] getNodes() {
+    	return nodes;
+    }
+
+    /**
+     * Set training data.
+     *
+     * @param docIndices Indices of selected documents
+     * @param words Document words
+     * @param labels Document labels
+     */
+    public void train(ArrayList<Integer> docIndices, int[][] words, int[][] labels) {
```

Comment:
So I don't think you are training a new model while the plugin is running? Should this logic be bundled with with plugin at all?